Lib/unittest/test/test_case.py

```diff
@@ -1673,6 +1673,18 @@ def testAssertLogsFailureLevelTooHigh(self):
                 with self.assertLogs(level='WARNING'):
                     log_foo.info("1")
 
+    def testAssertLogsFailureLevelTooHigh_FilterInRootLogger(self):
+        # Failure due to level too high - message propagated to root
+        with self.assertNoStderr():
+            oldLevel = log_foo.level
+            log_foo.setLevel(logging.INFO)
+            try:
+                with self.assertRaises(self.failureException):
+                    with self.assertLogs(level='WARNING'):
+                        log_foo.info("1")
+            finally:
+                log_foo.setLevel(oldLevel)
+
     def testAssertLogsFailureMismatchingLogger(self):
         # Failure due to mismatching logger (and the logged message is
```

Comment:
Note that this test is similar to the one just above it, which doesn't set the level on the log_foo logger. In that case, log_foo has level WARNING, so the message is blocked by it. But in the new test log_foo lets the message through to the root logger, which has that _CapturingHandler  attached to it. But since the _CapturingHandler's level was not set, it is NOTSET which is 0 so I guess it allows all messages through. So my change sets the level of the handler to the level specified by the user.

So with my change we set the log level both on the logger and on the handler attached to it. This seems to fix this particular bug but I'm not sure the interplay between the logger's and the handler's log level is always what we would expect it to be. Here is a small script demonstrating what's going on:

```
import logging

FORMAT = '%(levelname)-10s %(asctime)s: %(message)s'
handler = logging.StreamHandler()

# Uncomment the next line to fix it
# handler.setLevel(logging.WARNING) 
logging.basicConfig(format=FORMAT, level=logging.WARNING, handlers=[handler])

logger = logging.getLogger('blah')
logger.setLevel(logging.INFO)

logger.info('information')
logger.warning('warning')
logger.error('error')

```

As it is, all 3 log lines are emitted. If you uncomment the setLevel on the handler, only the warning and error are.